File: frontend/src/lib/types/dashboard-commerce.ts
export interface CommerceKPIs {
  revenue: number;
  revenue_change_pct: number | null;
  orders_count: number;
  orders_pending: number;
  units_sold: number;
  live_campaigns: number;
  draft_campaigns: number;
  avg_order_value: number;
}

export interface RevenueByDay {
  date: string;
  revenue: number;
  orders: number;
}

export interface OrdersByStatus {
  status: string;
  count: number;
}

export interface TopCampaignSummary {
  id: string;
  title: string;
  slug: string;
  status: string;
  units_sold: number;
  revenue: number;
}

export interface RecentOrderSummary {
  id: string;
  customer_email: string;
  total: number;
  status: string;
  created_at: string;
  campaign_title: string | null;
}

export interface StoreSummary {
  name: string;
  slug: string;
  has_logo: boolean;
  storefront_url: string;
}

export interface SetupStatus {
  has_logo: boolean;
  has_live_campaign: boolean;
  has_paid_order: boolean;
}

export interface CommerceDashboardOverview {
  period_start: string;
  period_end: string;
  kpis: CommerceKPIs;
  revenue_by_day: RevenueByDay[];
  orders_by_status: OrdersByStatus[];
  top_campaigns: TopCampaignSummary[];
  recent_orders: RecentOrderSummary[];
  store: StoreSummary | null;
  setup: SetupStatus;
}

File: backend/app/schemas/dashboard_commerce.py
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class CommerceKPIs(BaseModel):
    revenue: Decimal
    revenue_change_pct: Decimal | None = None
    orders_count: int
    orders_pending: int
    units_sold: int
    live_campaigns: int
    draft_campaigns: int
    avg_order_value: Decimal


class RevenueByDay(BaseModel):
    date: date
    revenue: Decimal
    orders: int


class OrdersByStatus(BaseModel):
    status: str
    count: int


class TopCampaignSummary(BaseModel):
    id: str
    title: str
    slug: str
    status: str
    units_sold: int
    revenue: Decimal


class RecentOrderSummary(BaseModel):
    id: str
    customer_email: str
    total: Decimal
    status: str
    created_at: datetime
    campaign_title: str | None = None


class StoreSummary(BaseModel):
    name: str
    slug: str
    has_logo: bool
    storefront_url: str


class SetupStatus(BaseModel):
    has_logo: bool
    has_live_campaign: bool
    has_paid_order: bool


class CommerceDashboardOverview(BaseModel):
    period_start: date
    period_end: date
    kpis: CommerceKPIs
    revenue_by_day: list[RevenueByDay]
    orders_by_status: list[OrdersByStatus]
    top_campaigns: list[TopCampaignSummary]
    recent_orders: list[RecentOrderSummary]
    store: StoreSummary | None = None
    setup: SetupStatus
